### DISCUSSION TITLE
How do setting pickup locations work?

### DISCUSSION THREAD

**Discussion by nabbl** [unchainedshop/unchained#351] [Q&A] [answered]
Hey there.

I need to be able to set a specific pickup location for each order. I added the Delivery provider "manual pickup" and created a configuration looking like that:

``` { key: "stores" , value: "[ { _id: 0, name: "bla", "address" : { .... }] ```

I then created an order and added this deliveryprovider to the order via this mutation: `mutation setDeliveryprovider`

After that I did this mutation `mutation updateOrderDeliveryPickUp("deliveryid: "something",  orderPickUpLocationId:"0"). ` using the provided IDs.

I can see that the order is updated correctly and I get no errors but activepickuplocation stays `null`

How exactly is this supposed to work?

**Comment by nabbl** [comment, marked as answer]
Ok I realized that setting 0 as an ID is not a good idea. Changed it to a random string and then it seems  to work.

**Reply by nabbl** [reply]
What happens afterwards is funny. When I finally submit the order it reverts the deliveryprovider of the order back to "SHIPPING" instead of using the "PICKUP" provider

**Reply by pozylon** [reply]
Can you try to change the _id to something else? _id in config seems to be a number but you provide it as string

Maybe you can identify the store in a string fashion to not fall for this. I hope we get to do some typescript type files soon.

**Reply by pozylon** [reply]
"0" !== 0

**Reply by pozylon** [reply]
Oh, I see you already found that out... so the submitted order has SHIPPING as type?! That seems quite unreal to me, sure it is the same order ?

**Reply by nabbl** [reply]
> Oh, I see you already found that out... so the submitted order has SHIPPING as type?! That seems quite unreal to me, sure it is the same order ?

Yes definitely the same order. I am going to try something again, maybe the issue is on my end

**Comment by pozylon** [comment]
@nabbl I can get to it tomorrow, sorry for the wait.
